src/coreclr/jit/morph.cpp

```diff
@@ -4685,12 +4972,64 @@ GenTree* Compiler::fgMorphMultiregStructArg(GenTree* arg, fgArgTabEntry* fgEntry
             //
             unsigned offset = baseOffset;
             newArg          = new (this, GT_FIELD_LIST) GenTreeFieldList();
+#if defined(TARGET_LOONGARCH64)
+            uint32_t numFloatFields = info.compCompHnd->getLoongArch64PassStructInRegisterFlags(objClass);
+            if (numFloatFields & (STRUCT_HAS_FLOAT_FIELDS_MASK ^ STRUCT_FLOAT_FIELD_ONLY_ONE))
+            {
+                assert((numFloatFields & STRUCT_HAS_FLOAT_FIELDS_MASK) > 1);
+                var_types tmp_type_1;
+                var_types tmp_type_2;
+
+                compFloatingPointUsed = true;
+                if (numFloatFields & STRUCT_FLOAT_FIELD_ONLY_TWO)
+                {
+                    tmp_type_1 = numFloatFields & STRUCT_FIRST_FIELD_SIZE_IS8 ? TYP_DOUBLE : TYP_FLOAT;
+                    tmp_type_2 = numFloatFields & STRUCT_SECOND_FIELD_SIZE_IS8 ? TYP_DOUBLE : TYP_FLOAT;
+                }
+                else if (numFloatFields & STRUCT_FLOAT_FIELD_FIRST)
+                {
+                    tmp_type_1 = numFloatFields & STRUCT_FIRST_FIELD_SIZE_IS8 ? TYP_DOUBLE : TYP_FLOAT;
+                    tmp_type_2 = numFloatFields & STRUCT_SECOND_FIELD_SIZE_IS8 ? type[1] : TYP_INT;
+                }
+                else if (numFloatFields & STRUCT_FLOAT_FIELD_SECOND)
+                {
+                    tmp_type_1 = numFloatFields & STRUCT_FIRST_FIELD_SIZE_IS8 ? type[0] : TYP_INT;
+                    tmp_type_2 = numFloatFields & STRUCT_SECOND_FIELD_SIZE_IS8 ? TYP_DOUBLE : TYP_FLOAT;
+                }
+                else
+                {
+                    assert(!"----------------unimplemented type-case... on LOONGARCH");
+                    unreached();
+                }
```

Comment:
This pattern seems to exist is multiple places. Would it make sense and be possible to create a helper function for this?